### PR TITLE
(5/10) RUM-2895 Create an abstract CompositionGroupMapper

### PR DESCRIPTION
**Pull request by xgouchet**
_This PR has been created automatically by splitting a large branch_

RUM-2895 Create an abstract CompositionGroupMapper

### CODE REVIEW

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/AbstractCompositionGroupMapper.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        parameters: Sequence<ComposableParameter>,
+        boxWithDensity: Box,
+        uiContext: UiContext
+    ): ComposeWireframe?
+
+    /**
+     * Converts a Compose [Color] value to a web String (e.g. #FF003E)
+     */
+    protected fun convertColor(color: Long): String? {
+        return if (color == 16L) {
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/AbstractCompositionGroupMapper.kt:54]: why this value is special?

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/reflection/ComposeReflection.kt [hunk 2]
```diff
@@ -8,6 +8,26 @@ package com.datadog.android.sessionreplay.compose.internal.reflection
 
 import java.lang.reflect.Field
 
+internal object ComposeReflection {
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/reflection/ComposeReflection.kt:11]: means we need to ship custom proguard rules with the library

**Review by 0xnm — DISMISSED** [review]

**Review by mariusc83 — APPROVED** [review]

**Review by jonathanmos — DISMISSED** [review]

File: features/dd-sdk-android-session-replay-compose/consumer-rules.pro [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# Keep the Compose internals class name. We need this in the SR recorder.
+-keepnames class androidx.appcompat.widget.DropDownListView
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/consumer-rules.pro:2]: it seems we should have more classes here by looking on the content of `ComposeReflection`

**Review by 0xnm — DISMISSED** [review]
lgtm! the only concern is the content of Proguard rules file, it needs to be updated.

**Review by 0xnm — APPROVED** [review]

**Pull request merged**